.github/workflows/continuous_integration.yml

```diff
@@ -0,0 +1,170 @@
+            openpbta_subset: 0
+
+          - name: Molecular Subtyping - EWS
+            entrypoint: molecular-subtyping-EWS/run_subtyping.sh
+
+          - name: Molecular Subtyping - NEUROCYTOMA
+            entrypoint: molecular-subtyping-neurocytoma/run_subtyping.sh
+
+          - name: Molecular Subtyping - HGG
+            entrypoint: molecular-subtyping-HGG/run-molecular-subtyping-HGG.sh
+            openpbta_subset: 0
+
+          - name: Molecular Subtyping - LGG
+            entrypoint: molecular-subtyping-LGAT/run_subtyping.sh
+            openpbta_subset: 0
+
+          - name: Molecular Subtyping - PATHOLOGY/COMPILE
+            entrypoint: molecular-subtyping-pathology/run-subtyping-aggregation.sh
+            openpbta_testing: 1
+
+          - name: Molecular Subtyping - INTEGRATE (+ add cancer groups)
+            entrypoint: molecular-subtyping-integrate/run-subtyping-integrate.sh
+
+          # Analysis modules
+          - name: Independent Specimens
+            entrypoint: independent-samples/run-independent-samples.sh
+
+          - name: Independent Specimens pre-release
+            entrypoint: independent-samples/run-independent-samples.sh
+            run_for_subtyping: 1
+
+          - name: TP53/NF1 scores
+            entrypoint: tp53_nf1_score/run_classifier.sh
+            openpedcan_polya_strand: 0
+
+          - name: Fusion filtering
+            entrypoint: fusion_filtering/run_fusion_merged.sh
+
+          - name: Fusion summary
+            entrypoint: fusion-summary/run-new-analysis.sh
+            openpbta_subset: 0
+
+          - name: Consensus CN
+            entrypoint: focal-cn-file-preparation/run-prepare-cn-OpenTarget.sh
+            openpbta_testing: 1
+
+          - name: TMB calculation
+            entrypoint: tmb-calculation/run_tmb_calculation.sh
+
+          - name: Immune Deconvolution
+            entrypoint: immune-deconv/run-immune-deconv.sh
+
+          - name: EFO/MONDO annotation
+            entrypoint: efo-mondo-mapping/run_qc.sh
+
+          - name: ENSEMBL Gene matching
+            entrypoint: gene_match/run-gene-mapping.sh
+
+          - name: Update table annotation data
+            entrypoint: long-format-table-utils/run-update-long-format-table-utils.sh
+
+          - name: Fusion frequency tables
+            entrypoint: fusion-frequencies/run-frequencies.sh
+    steps:
+      - uses: actions/checkout@v3
+
+      - name: Cache Data
+        uses: actions/cache@v3
+        with:
+          path: ./data
+          key: ${{ runner.os }}-data-${{ hashFiles('download-data.sh') }}
+          restore-keys: |
+            ${{ runner.os }}-data-
+
+      - name: Download Data
+        uses: docker://pgc-images.sbgenomics.com/d3b-bixu/open-pedcan:analysisjob
+        with:
+          entrypoint: ./download-data.sh
+        env:
+          OPENPEDCAN_URL: https://s3.amazonaws.com/d3b-openaccess-us-east-1-prd-pbta/open-targets
+          OPENPEDCAN_RELEASE: testing
```

Comment:
We could wrap the entire `run_analysis` job in the build container. However, the upstream [`rocker/tidyverse`](https://hub.docker.com/r/rocker/tidyverse) image is based on an old version of Debian that doesn't contain Git 2.18+. Without Git 2.18+ available on the PATH, GitHub Actions will download an archive of the repo via HTTP. The analyses depend on the `.git` folder being present. So, short of getting a newer version of Git into the build container, we run the checkout step outside of the container and specifically use the container in these two steps.

Reply:
@jharenza is there a ticket open for evaluating new R images?

Reply:
There is not, but we can add one. @brianghig had noticed over in the [api repo ](https://github.com/PediatricOpenTargets/OpenPedCan-api/pull/77) that updating this image sped up building the docker image as well.

I do not think we should do that update here, and I presume many analyses might break with an update, so will have to test carefully in a separate PR.